tests/test_api.py

```diff
@@ -981,14 +985,140 @@ def test_empty_to_uuid(self):
             api.move_identity('03877f31261a6d1a1b3971d240e628259364b8ac', '')
 
 
+class TestAddOrganization(TestCase):
+    """Unit tests for add_organization"""
+
+    def test_add_new_organization(self):
+        """Check if everything goes OK when adding a new organization"""
+
+        organization = api.add_organization(name='Example')
+
+        # Tests
+
+        self.assertIsInstance(organization, Organization)
+
+        self.assertEqual(organization.name, 'Example')
+
+        organizations_db = Organization.objects.filter(name='Example')
+        self.assertEqual(len(organizations_db), 1)
+
+        org1 = organizations_db[0]
+
+        self.assertEqual(organization, org1)
+
+    def test_add_duplicate_organization(self):
+        """Check if it fails when adding a duplicate organization"""
+
+        org = api.add_organization(name='Example')
+
+        with self.assertRaisesRegex(AlreadyExistsError, ORGANIZATION_ALREADY_EXISTS_ERROR.format(name=org.name)):
+            org = api.add_organization(name=org.name)
+
+        organizations = Organization.objects.filter(name='Example')
```

Comment:
This test is good because it checks if it didn't mess things up when adding the same organization twice. It will also better if it checks the number of total organizations.

Reply:
Done.